core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java

```diff
@@ -153,17 +156,37 @@ private PreparedStatement prepareQueryWithColunmsReturned(Connection con, String
 		return con.prepareStatement(query,columnsReturned);
 	}
 
-	protected PreparedStatement prepareQuery(Connection con, String query, boolean updateable) throws SQLException {
+	protected PreparedStatement prepareQuery(Connection con, String query, boolean updateable) throws SQLException, JdbcException {
+		String convertedQuery = convertQuery(con, query.trim());
 		if (log.isDebugEnabled()) {
-			log.debug(getLogPrefix() +"preparing statement for query ["+query+"]");
+			log.debug(getLogPrefix() +"preparing statement for query ["+convertedQuery+"]");
 		}
 		String[] columnsReturned = getColumnsReturnedList();
 		if (columnsReturned!=null) {
-			return prepareQueryWithColunmsReturned(con,query,columnsReturned);
+			return prepareQueryWithColunmsReturned(con,convertedQuery,columnsReturned);
 		}
-		return con.prepareStatement(query,ResultSet.TYPE_FORWARD_ONLY,updateable?ResultSet.CONCUR_UPDATABLE:ResultSet.CONCUR_READ_ONLY);
+		return con.prepareStatement(convertedQuery,ResultSet.TYPE_FORWARD_ONLY,updateable?ResultSet.CONCUR_UPDATABLE:ResultSet.CONCUR_READ_ONLY);
 	}
 
+	protected String convertQuery(Connection connection, String query) throws JdbcException, SQLException {
+		if (StringUtils.isNotEmpty(sqlDialect)) {
```

Comment:
Test zou moeten zijn: 
if (StringUtils.isNotEmpty(sqlDialect) && !sqlDialect.equals(getDbmsSupport().getSqlDialect()))

waarbij getDbmsSupport().getSqlDialect() in eerste instantie gelijk is aan getDbmsSupport().getDbmsName(),
maar wie weet.

Reply:
https://github.com/ibissource/iaf/pull/251/commits/8f3dcc947d1dda5bc788cad33e341ff8e1890e15